README.md

```diff
@@ -259,6 +259,42 @@ OrderState.values # ['CREATED', 'PAID']
 ShippedOrderState.values # ['CREATED', 'PAID', 'PREPARED', SHIPPED']
 ```
 
+### Exhaustive case matcher
+
+If you want to make sure that you cover all cases in a case stament, you can use the exhaustive case matcher: `Ruby::Enum::Ecase`. It will raise an error if a case/enum value is not handled or if a value is specified that's not part of the enum. This is inspired by the [Rust Pattern Syntax](https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html).
+
+> NOTE: This will add checks at runtime which might lead to slightly worse performance.
```

Comment:
I am not sure this is needed. All code has performance impact. Unless you believe it's slower by a factor?

Reply:
The thing is that a normal case statement is pretty fast I would say. However, this enum case statement creates some lists in between and checks on them for every execution. That's why I would assume it's slower by a factor from what one would use normally. But I haven't performance tested it.

Should I still remove it? Or should I add a further explanation?

Reply:
I just tested it with the following code:

```ruby
  describe 'performance' do
    let(:cases) do
      {
        [test_enum::RED, test_enum::GREEN] => -> { 'red or green' },
        test_enum::BLUE => -> { 'blue' }
      }
    end

    it 'is faster than a regular case statement' do
      ruby_enum_time = Benchmark.realtime do
        1_000_000.times do
          test_enum.case(test_enum::RED, cases)
        end
      end

      case_statement_time = Benchmark.realtime do
        1_000_000.times do
          case test_enum::RED
          when test_enum::RED, test_enum::GREEN
            'red or green'
          when test_enum::BLUE
            'blue'
          end
        end
      end

      puts "ruby-enum: #{ruby_enum_time.round(4)}"
      puts "case statement: #{case_statement_time.round(4)}"

      puts "ruby-enum is #{(ruby_enum_time / case_statement_time).round(2)} times slower"
    end
  end
```

And the result is bad, so the word `slightly` should be removed or replaced with `a lot`

```ruby
ruby-enum: 2.4736
case statement: 0.0774
ruby-enum is 31.98 times slower
```

Reply:
I might look into ways to improve its performance ...

Reply:
(I used Ruby 3.1 and Ruby 3.2 is similar)

Reply:
Nothing is free in meta programming :) Let's leave the comment! You should contribute these benchmarks as well into a new `benchmarks` subfolder with a quick how to run them README in a separate PR.

Reply:
Good idea. Done :)